spec/policies/person_policy_spec.rb

```diff
@@ -157,3 +122,331 @@
 
   end
 end
+
+describe PersonPolicy, "given an unlinked user" do
+  let(:user) do
```

Comment:
I think we should be testing a real user with consumer role; Sai created a good test on https://github.com/ideacrew/enroll/pull/3633/files that invloves ridp verified, but its basically use a factory 

let(:person) { FactoryBot.create(:person, :with_consumer_role, :with_active_consumer_role) }
let(:consumer_role) { person.consumer_role }

and then 
consumer_role.update_attributes(identity_validation: 'rejected')

Reply:
Interacting with the DB is very slow, comparatively.  Using factories when we don't need to is one of the reason's we have 20 minute test runs compared to similar size projects.

We're fine to mock, under very certain circumstances, provided we do it the right way:
1.  Try to only mock what you own.
2. Use `instance_double` rather than `double`.  RSpec verifies the mocks by preventing you from calling methods to which the 'real' type would never respond.
3. Mock things for which the corresponding behaviour has solid testing.  In this case, I am confident the objects I am mocking have tests, such as Person.primary_family, User.person, etc.
4. You need to, at some point, be able to 'trust the framework' - I have to be able to assume at a certain point that my objects can act in isolation, that associations work as I expect.

However:
1. We definetly have a code smell based on the QUANTITY and DEPTH of the mocks required.
2. The primary symptom of this should be that I have methods on the database objects in question which allow getting at objects without quite as much demeter violation - and **those** methods need to both:
  1. Have specs
  2. Be made use of in `ApplicationPolicy`.